libs/akita/src/__tests__/entityStoreCache.spec.ts

```diff
@@ -0,0 +1,158 @@
+      { id: 1, token: 'token 1' },
+      { id: 2, token: 'token 2' },
+    ]);
+
+    expect(store._value().ids.length).toBe(2);
+    expect(store._value().idsExpired.length).toBe(0);
+
+    await new Promise((accept) => setTimeout(accept, 2000));
+    expect(store._value().ids.length).toBe(2);
+    expect(store._value().idsExpired).toContain(1);
+    expect(store._value().idsExpired).toContain(2);
+  });
+
+  it('updated entity should expire later', async () => {
+    store = new TestAuthStore(
+      {},
+      {
+        cache: {
+          ttl: 1000,
+        },
+      }
+    );
+
+    const base = Date.now();
+    store.expired$.subscribe(({ id }) => {
+      if (id === 1) {
+        expect(Date.now()).toBeGreaterThan(base + 1500);
+      }
+      if (id === 2) {
+        expect(Date.now()).toBeGreaterThan(base + 1000);
+      }
+    });
+
+    store.set([
+      { id: 1, token: 'token 1' },
+      { id: 2, token: 'token 2' },
+    ]);
+
+    expect(store._value().ids.length).toBe(2);
+    expect(store._value().idsExpired.length).toBe(0);
+
+    await new Promise((accept) => setTimeout(accept, 500)); //  500ms
+    store.update(1, { token: 'token 1 new' });
+    expect(store._value().ids.length).toBe(2);
+    expect(store._value().idsExpired.length).toBe(0);
+
+    await new Promise((accept) => setTimeout(accept, 500)); // 1000ms
+    expect(store._value().ids.length).toBe(2);
+    expect(store._value().idsExpired).not.toContain(1);
+    expect(store._value().idsExpired).toContain(2);
+
+    await new Promise((accept) => setTimeout(accept, 1000)); // 2000ms
+    expect(store._value().ids.length).toBe(2);
+    expect(store._value().idsExpired).toContain(1);
+    expect(store._value().idsExpired).toContain(2);
+  });
+
+  it('ttlEntityCache', async () => {
+    store = new TestAuthStore(
+      {},
+      {
+        cache: {
+          ttl: 1000,
+        },
+      }
+    );
+
+    store.set([
+      { id: 1, token: 'token 1' },
+      { id: 2, token: 'token 2' },
+    ]);
+
+    expect(store._value().ids.length).toBe(2);
+
+    const source = of({ id: 1, token: 'token fetched' } as TestAuth);
+
+    {
+      const result = await source.pipe(ttlEntityCache(store, 1), first()).toPromise();
+      expect(result.id).toBe(1);
+      expect(result.token).toBe('token 1');
+    }
+
+    await new Promise((accept) => setTimeout(accept, 1500));
+
+    {
+      const result = await source.pipe(ttlEntityCache(store, 1), first()).toPromise();
+      expect(result.id).toBe(1);
+      expect(result.token).toBe('token fetched');
+    }
+  });
+
+  it('useEntityCache', async () => {
+    store = new TestAuthStore(
+      {},
+      {
+        cache: {
+          invalidation: {
+            entity: (entity: TestAuth) => entity.token === undefined,
+          },
+        },
+      }
+    );
+
+    store.set([
+      { id: 1, token: undefined },
+      { id: 2, token: 'token 2' },
+    ]);
+
+    expect(store._value().ids.length).toBe(2);
+
+    {
```

Comment:
Where do you test that it doesn't invoke the source when it's cached?

Reply:
The code block below: https://github.com/datorama/akita/pull/471/files/8f24df0e47610bafbe4735e095d7d4c3dd074ba1#diff-de1efd0295ddb2cc00b19d5c3f6ec4f4R155